src/v/pandaproxy/json/requests/fetch.h

```diff
@@ -42,20 +42,26 @@ class rjson_serialize_impl<model::record> {
       , _tpv(tpv)
       , _base_offset(base_offset) {}
 
-    void
+    bool
     operator()(::json::Writer<::json::StringBuffer>& w, model::record record) {
         w.StartObject();
         w.Key("topic");
         ::json::rjson_serialize(w, _tpv.topic);
         w.Key("key");
-        rjson_serialize_fmt(_fmt)(w, record.release_key());
+        if (!rjson_serialize_fmt(_fmt)(w, record.release_key())) {
+            throw serialize_error("Unable to serialize key value of record");
```

Comment:
It's not clear to me whether we should return false or throw. One advantage of throwing is that the message can contain context such as an offset.

Reply:
The reason I'm throwing here is because this is being called from within `for_each_record` on lines 104 - 107.  That method doesn't return anything so I can't propagate the error outside of it.